saas/renewals.py

```diff
@@ -116,11 +116,26 @@ def _recognize_subscription_income(subscription, until=None):
                         'nb_periods': nb_periods,
                         'period_start': descr_period_start,
                         'period_end': descr_period_end})
+
             recognize_period_idx += 1
             recognize_start = (subscription.created_at
                 + relativedelta(months=recognize_period_idx))
             recognize_end = (subscription.created_at
                 + relativedelta(months=recognize_period_idx + 1))
+        use_charges = subscription.plan.use_charges.all()
+        use_charge_amount = 0
+        for use_charge in use_charges:
+            quantity = get_period_usage(subscription, use_charge,
+                recognize_start, recognize_end)
+            extra = quantity - use_charge.quota
+            if extra > 0:
+                use_charge_amount += extra * use_charge.use_amount
+        if use_charge_amount > 0:
+            descr = "use charge for plan %d" % subscription.plan.id
+            Transaction.objects.create_income_recognized(
```

Comment:
`create_income_recognized` is special in that it is designed to recognize Income on a subscription by taking it out of the backlog (or receivable if necessary). See code samples from it:

    balance = self.get_event_balance(subscription.id,                # XXX Here replace subscription.id by get_event_id
            account=Transaction.BACKLOG, ends_at=ends_at)

over-quota use charges haven't been invoiced nor paid yet (as opposed to the subscription to plan). We want to record the following transactions (example):

    2018/09/09 _use charge event_                                  ; Now "past due" since the service was delivered.
        subscriber:Liability            _over quota amount_
        subscriber:Payable
       ; event_id=sub_xx_yy
       
    2018/09/09 _use charge event_  income recognized   ; Since the service was delivered before payment.
        provider:Receivable           _over quota amount_
        provider:Income
       ; event_id=sub_xx_yy

There might need for another Transaction to balance the ledger once the Charge.payment_successful is called.

Reply:
The 2 transaction seems correct (crediting income (revenue) and debiting receivable), however the first one seems to be wrong because payable is a liability too, wouldn't it make sense to debit (+) provider's receivable and credit (-) subscriber payable?

Reply:
Now that I'm thinking about this (the first transaction): we already log transactions on each event, so there is no need for 2 extra transactions in the end, a single one (that recognizes income and debits receivable) would be enough, isn't it?

Reply:
And once a user pays (e.g. Charge.payment_successful) we create another transaction (credit receivable and debit cash account (is it named funds in djaodjin?))